### ISSUE TITLE
Why is 'Parse error' a warning

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#182]
When HLint encounters an error parsing the source it reports this as a warning with the output `File.hs:123:45: Warning: Parse error: message`

Would it be more appropriate to report this as an error. I think that unparsable code is a /strong/ error in style ;)

**Comment by ndmitchell** [comment]
The idea is that you are presumably running your code through a Haskell compiler and running it before using HLint, or at the very least through an editor with smart feedback or [ghcid](https://github.com/ndmitchell/ghcid). Therefore, getting a parse error means HLint is in the wrong, not you - it might have the wrong CPP symbols in scope, you might be using a feature that the haskell-src-exts parser gets wrong etc. It's less likely to be a real style error.

**Comment by ndmitchell** [comment]
Once we have agreed on an "answer", this should go into the FAQ.

**Comment by expipiplus1** [comment]
What made me notice this was running HLint alongside ghc-mod in vim with Neomake. Neomake was reporting warnings for a file which wouldn't even compile which I thought was a little odd.

HLint and ghc-mod are run in parallel, so there's no (easy) way to remove HLint output if ghc-mod returns an error (nor would I want to because hlint messages are still useful even in the presence of type errors for example)

I'm currently treating output from HLint containing `Warning: Parse error` as an error in my fork of neomake which solves this for me, so that might be the best solution.

**Comment by ndmitchell** [comment]
Fair enough. Note that you can get the parse error hint to be an error with:

    $ hlint Sample.hs --with="error """Parse error""" "
    Sample.hs:2:16: Error: Parse error: !
    Found:
        {-# LANGUAGE BangPatterns, ViewPatterns #-}
        someFun (id -> !arg) = undefined
      >

That's on Windows. On Linux I imagine the more sane `--with="error \"Parse error\"" will work.

**Comment by expipiplus1** [comment]
That looks even better, thanks!

**Comment by expipiplus1** [comment]
Sorry, probably shouldn't close this until the faq is updated.

**Comment by jasonstolaruk** [comment]
Hello, the project I'm working on has 6 suggestions that are parse errors (I think the number jumped from 2 to 6 after I upgraded to version 1.9.25). If parse errors indicate that HLint is in the wrong, would you like me to open a new issue here to address them?

**Comment by ndmitchell** [comment]
@jasonstolaruk yes please! That new version corresponds to a new version of the parser library, so probably a regression. If the project is public just list the locations of the files and I'll take it from there.

**Comment by ndmitchell** [comment]
As part of #99, I decided that saying "Suggestion: Parse error" is just too silly, so I've made parse errors severity error.
